### DISCUSSION TITLE
Detect orientation of PDF and rotate accordingly?

### DISCUSSION THREAD

**Discussion by CaiSamuelsFSA** [pymupdf/PyMuPDF#3254] [General]
This is probably an obvious question but I am extracting contents from multiple PDF files and some of them are landscape orientation. Using my current script, I get a KeyError exception when trying to read tables from PDFs that are landscape orientation and I assume this is because I haven't rotated the document. How do I automatically detect the orientation and rotate the document accordingly?

**Comment by JorjMcKie** [comment]
Which version do you use? At a certain version, I am internally de-rotating a page before table detection.

**Reply by CaiSamuelsFSA** [reply]
1.23.15

**Reply by JorjMcKie** [reply]
Ah yes, I think that one still had the problem. Try to upgrade please.

**Reply by CaiSamuelsFSA** [reply]
So with newer versions there is no need to rotate a landscape document before text and table extraction?
